lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -259,6 +259,131 @@ class ExportCommand < HammerCLIKatello::SingleResourceCommand
+      )
+
+      validate_options do
+        option(:option_export_dir).required
+      end
+
+      build_options
+
+      class ContentViewIdParamSource
+        def initialize(command)
+          @command = command
+        end
+
+        def get_options(_defined_options, result)
+          if result['option_id'].nil? && result['option_name']
+            result['option_id'] = @command.resolver.content_view_id(result)
+          end
+          result
+        end
+      end
+
+      def option_sources
+        sources = super
+        idx = sources.index { |s| s.class == HammerCLIForeman::OptionSources::IdParams }
+        sources.insert(idx, ContentViewIdParamSource.new(self))
+        sources
+      end
+
+      def execute
+        cvv = content_view_version
+        repositories = cvv['repositories'].collect do |repo|
+          repository(repo['id'])
+        end
+
+        repositories.each do |repo|
+          repo['packages'] = packages(repo['id'])
+          repo['errata'] = errata(repo['id'])
+        end
+
+        json = export_json(cvv, repositories)
+        create_tar(cvv, repositories, json)
+      end
+
+      def create_tar(cvv, repositories, json)
+        export_file = "export-#{cvv['id']}.json"
+        export_repos_tar = "exported-repos-#{cvv['id']}.tar"
+        export_tar = "export-#{cvv['id']}.tar"
+        published_repo_dir = "/var/lib/pulp/published/yum/https/repos/"
+
+        Dir.chdir(published_repo_dir) do
+          repositories.each do |repo|
+            `tar cvfh #{options['option_export_dir']}/#{export_repos_tar} #{repo['relative_path']}`
+          end
+        end
+
+        Dir.chdir(options['option_export_dir']) do
+          File.open(export_file, 'w') do |file|
+            file.write(json.to_json)
+          end
+
+          `tar cf #{export_tar} #{export_repos_tar} #{export_file}`
+
+          File.delete(export_repos_tar)
+          File.delete(export_file)
+        end
+      end
+
+      def content_view_version
+        resource_content_view_versions.call(:show, 'id' => options['option_id'])
+      end
+
+      def resource_content_view_versions
+        HammerCLIForeman.foreman_resource(:content_view_versions)
+      end
+
+      def repository(id)
+        HammerCLIForeman.foreman_resource(:repositories).call(:show, 'id' => id)
+      end
+
+      def packages(id)
+        HammerCLIForeman.foreman_resource(:packages).call(:index, 'repository_id' => id)['results']
+      end
+
+      def errata(id)
+        HammerCLIForeman.foreman_resource(:errata).call(:index, 'repository_id' => id)['results']
+      end
+
+      def export_json(content_view_version, repositories)
+        json = {
+          "name" => content_view_version['content_view']['name'],
+          "major" => content_view_version['major'],
+          "minor" => content_view_version['minor']
+        }
+
+        json["repositories"] = repositories.collect do |repo|
+          {
+            "repo_id" => repo['id'],
```

Comment:
Why the `repo_` prefix? IMO these fields are all repo fields if they're in an array pointed to by the "repositories" key. We don't do this in Katello; what's the motivation here?

Reply:
This was following the original design by @beav . I do agree I think this can be simplified. Will let him weigh in